TakingITGlobal/commit2act
# TIG-761 - AvatarSelectTile and its AvatarSelectGroup counterpart component

# Description
[TIG-761 - Avatar Select Tile](https://app.clickup.com/t/9009201449/TIG-761)

In this PR, I created the AvatarSelectTile component. This component will be used in the Sign-Up flow when the user is selecting their avatar. 

The success criteria for this PR is:

- [ ] Can tab into the radio group, highlighting the first (or selected) option
- [ ] Can use the arrow keys to navigate between each option
- [ ] Can click either the image or title for each avatar card to select it
- [ ] Can see a default option selected when initially viewing the component(s)

This PR has the images as placeholders (all avatar images are the same).

EDIT: The Component is now renamed to SingleSelectTile. The component can be focused on using tab and the choice can be selected using Enter or space

# Testing

1. Pull down and install the branch (`npm install`)
2. Run all tests (`npm run test:basic`) and confirm they pass
3. Run Storybook (`npm run storybook`)
4. Under `AvatarSelectTile` make sure the `Unchecked` version you CAN see the avatar
5. Under `AvatarSelectTile` make sure the `Unchecked` version you CANNOT see the avatar (grey checkmark)
6. Under `AvatarSelectGroup` make sure the first avatar is checked by default. Make sure the selection can be selected using tab and arrow keys.

# Images
![image](https://github.com/TakingITGlobal/commit2act/assets/132935720/31992b85-92bc-4332-881f-061404fc4232)
![image](https://github.com/TakingITGlobal/commit2act/assets/132935720/923e3440-6d76-44e0-8915-fcc93951be91)
![image](https://github.com/TakingITGlobal/commit2act/assets/132935720/1dffb3da-eba3-42b8-b5d6-a84a67d6194f)
